plumbing/format/idxfile/decoder.go

```diff
@@ -17,18 +17,17 @@ var (
 	ErrMalformedIdxFile = errors.New("Malformed IDX file")
 )
 
-// A Decoder reads and decodes idx files from an input stream.
+// Decoder reads and decodes idx files from an input stream
 type Decoder struct {
 	io.Reader
 }
 
-// NewDecoder returns a new decoder that reads from r.
+// NewDecoder construct a new idx stream decoder
 func NewDecoder(r io.Reader) *Decoder {
 	return &Decoder{r}
 }
 
-// Decode reads the whole idx object from its input and stores it in the
-// value pointed to by idx.
+// Decode reads from the stream and decode the content into the Idxfile struct
```

Comment:
From what stream?, improve the ctor comment so this sentence makes sense.

End the comment with a full stop.